controller/script/simulate-proxy/main.go

```diff
@@ -119,8 +211,79 @@ var (
 	}
 )
 
-func randomPort() uint32 {
-	return ports[rand.Intn(len(ports))]
+var (
+	labels        = generatePromLabels()
+	requestTotals = prom.NewCounterVec(
+		prom.CounterOpts{
+			Name: "request_total",
+			Help: "A counter of the number of requests the proxy has received",
+		},
+		labels,
+	)
+	responseTotals = prom.NewCounterVec(
+		prom.CounterOpts{
+			Name: "response_total",
+			Help: "A counter of the number of responses the proxy has received.",
+		},
+		labels,
+	)
+	responseDurationMs = prom.NewHistogramVec(
+		prom.HistogramOpts{
+			Name:    "response_duration_ms",
+			Help:    "A histogram of the duration of a response",
+			Buckets: latencyBucketBounds,
+		}, labels)
+
+	responseLatenciesMs = prom.NewHistogramVec(
+		prom.HistogramOpts{
+			Name:    "response_latency_ms",
+			Help:    "A histogram of the total latency of a response.",
+			Buckets: latencyBucketBounds,
+		}, labels)
+
+	kubeResourceTypes = []string{
+		"job",
+		"replica_set",
+		"deployment",
+		"daemon_set",
+		"replication_controller",
+		"namespace",
+	}
+)
+
+func generatePromLabels() []string {
+	constantLabels := []string{
+		"direction",
+		"authority",
+		"status_code",
+		"grpc_status_code",
+		"instance",
```

Comment:
we should get `instance` for free from prometheus, it should be of the form `simulate-proxy:9000`